controllers/machinehealthcheck_targets.go

```diff
@@ -228,7 +228,11 @@ func (r *MachineHealthCheckReconciler) getNodeFromMachine(clusterClient client.R
 		Name: machine.Status.NodeRef.Name,
 	}
 	err := clusterClient.Get(context.TODO(), nodeKey, node)
-	return node, err
+	// if it cannot find a node, send a nil node back...
```

Comment:
This didn't seem correct.
`getNodeFromMachine` should return a nil node if it cannot find any node because that nil check happens in the `needRemediation` method for the healthCheckTarget.

Reply:
cc @JoelSpeed

Reply:
I think was a hangover from an original version where if the node wasn't found, we would see a non-nil node with no UID which would suggest the node no longer exists, this was replaced by `nodeMissing` as a field on the target struct.

I think therefore, that this change is fine :)

Reply:
Thanks for the confirmation @JoelSpeed!